server/src/main/java/org/solarbank/server/service/NasaClient.java

```diff
@@ -0,0 +1,55 @@
+package org.solarbank.server.service;
+
+import java.time.Duration;
+import java.util.Map;
+import org.solarbank.server.ErrorMessage;
+import org.solarbank.server.dto.Location;
+import org.solarbank.server.dto.NasaResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+@Service
+public class NasaClient {
```

Comment:
This is the Class for the WebClient that calls the Nasa API. I've made this a @Service, but now think maybe it should be a @Component?